src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/Triggers/DataBoxEdgeTriggerNewCmdletBase.cs

```diff
@@ -0,0 +1,250 @@
+
+        private const string PeriodicTimerTriggerParameterSet = "PeriodicTimerTriggerParameterSet";
+        private const string PeriodicTimerTriggerResourceIdParameterSet = "PeriodicTimerTriggerResourceIdParameterSet";
+
+        [Parameter(Mandatory = true, ParameterSetName = PeriodicTimerTriggerParameterSet,
+            HelpMessage = Constants.ResourceGroupNameHelpMessage,
+            Position = 0)]
+        [Parameter(Mandatory = true, ParameterSetName = FileEventTriggerParameterSet,
+            HelpMessage = Constants.ResourceGroupNameHelpMessage,
+            Position = 0)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = FileEventTriggerParameterSet,
+            HelpMessage = Constants.DeviceNameHelpMessage,
+            Position = 1)]
+        [Parameter(Mandatory = true, ParameterSetName = PeriodicTimerTriggerParameterSet,
+            HelpMessage = Constants.DeviceNameHelpMessage,
+            Position = 1)]
+        [ResourceNameCompleter("Microsoft.DataBoxEdge/dataBoxEdgeDevices", nameof(ResourceGroupName))]
+        [ValidateNotNullOrEmpty]
+        public string DeviceName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = FileEventTriggerParameterSet,
+            HelpMessage = Constants.NameHelpMessage,
+            Position = 2)]
+        [Parameter(Mandatory = true, ParameterSetName = PeriodicTimerTriggerParameterSet,
+            HelpMessage = Constants.NameHelpMessage,
+            Position = 2)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = PeriodicTimerTriggerParameterSet,
+            HelpMessage = HelpMessageTrigger.SinkInfoHelpMessage)]
+        [Parameter(Mandatory = true, ParameterSetName = FileEventTriggerParameterSet,
+            HelpMessage = HelpMessageTrigger.SinkInfoHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string RoleName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = PeriodicTimerTriggerResourceIdParameterSet,
+            HelpMessage = HelpMessageTrigger.SinkInfoHelpMessage)]
+        [Parameter(Mandatory = true, ParameterSetName = FileEventTriggerResourceIdParameterSet,
+            HelpMessage = HelpMessageTrigger.SinkInfoHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        private string RoleId { get; set; }
+
+
+        [Parameter(Mandatory = true,
+            ParameterSetName = FileEventTriggerParameterSet,
+            HelpMessage = HelpMessageTrigger.FileEventSwitchParameter)]
+        [Parameter(Mandatory = true,
+            ParameterSetName = FileEventTriggerResourceIdParameterSet,
+            HelpMessage = HelpMessageTrigger.FileEventSwitchParameter)]
+        [ValidateNotNullOrEmpty]
+        public SwitchParameter FileEvent { get; set; }
+
+        [Parameter(Mandatory = true,
```

Comment:
This is conflict with the design issue: https://github.com/raviten/azure-powershell/blob/raviten/databoxedge-cmdlets-ii/src/DataBoxEdge/DataBoxEdge/help/New-AzDataBoxEdgeTrigger.md#fileeventtriggerparameterset-default